File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var sumOddLengthSubarrays = function(arr) {
  let sum = 0;
  for (let i = 0; i < arr.length; i++) {
    for (let j = 0; j < arr.length; j++) {
      const subArray = arr.slice(i, j + 1);
      if (subArray.length % 2 !== 0) {
        sum += subArray.reduce((a, b) => a + b, 0);
      }
    }
  }
  return sum;
};

File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.py
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        total = 0
        l = len(arr)
        for i in range(l):
            for j in range(i, l, 2):
                total = total + sum(arr[i:j+1])
                
        return total
                